web/src/main/java/org/openmrs/web/filter/GZIPFilter.java

```diff
@@ -94,7 +119,35 @@ private boolean isGZIPEnabled() {
 		}
 		catch (Throwable t) {
 			log.warn("Unable to get the global property: " + OpenmrsConstants.GLOBAL_PROPERTY_GZIP_ENABLED, t);
-			// not caching the enabled flag here in case it becomes available 
+			// not caching the enabled flag here in case it becomes available
+			// before the next request
+			
+			return false;
+		}
+	}
+	
+	/**
+	 * Returns true if path matches pattern in gzip.acceptCompressedRequestsForPaths property
+	 */
+	private boolean isCompressedRequestForPathAccepted(String path) {
+		if (cachedCompressedRequestForPathAccepted != null)
+			return cachedCompressedRequestForPathAccepted;
+		
+		try {
+			String gzipCompressedRequestForPathAccepted = Context.getAdministrationService().getGlobalProperty(
+			    OpenmrsConstants.GLOBAL_PROPERTY_GZIP_ACCEPT_COMPRESSED_REQUESTS_FOR_PATHS, "");
+			cachedCompressedRequestForPathAccepted = false;
+			for (String acceptPath : gzipCompressedRequestForPathAccepted.split(",")) {
+				if (path.matches(acceptPath)) {
+					cachedCompressedRequestForPathAccepted = true;
```

Comment:
I see another issue here. You can't cache the result, because it will be different for each path. You should cache the global_property value instead.